lua/starfall/sflib.lua

```diff
@@ -139,6 +139,14 @@ SF.StructWrapper = {
 			end,
 			__metatable = name,
 			__tostring = function() return name end,
+			__printtable = function()
+				for k, v in pairs(data) do
+					if not cache[k] then
+						cache[k] = instance.WrapObject(v)
+					end
+				end
+				instance.env.printTable(cache)
```

Comment:
This is a security hole. Don't trust anything in `instance.env`

Reply:
You can set `SF.PrintTable = PrintTableX` and use SF.PrintTable instead.

Reply:
Nvm about the security hole comment, I didn't realize `cache` was all wrapped objects.

Reply:
`SF.PrintTable` would still be better

Reply:
I would do SF.PrintTable = printTableX, but I feel like the indentation would get messed up unless you want me to also pass in the indent count in the __printtable function.

Reply:
I feel like I could just make __printtable return a table to be printed in place of the original.

Reply:
Yeah you might want to do that. Also there's two `printTable`. It would be better if your __printtable modification was in PrintTableX instead as well.